libraries/AP_Motors/AP_MotorsMulticopter.cpp

```diff
@@ -339,13 +339,13 @@ float AP_MotorsMulticopter::get_current_limit_max_throttle()
 float AP_MotorsMulticopter::apply_thrust_curve_and_volt_scaling(float thrust) const
 {
     float throttle_ratio = thrust;
-    // apply thrust curve - domain 0.0 to 1.0, range 0.0 to 1.0
+    // apply thrust curve - domain -1.0 to 1.0, range -1.0 to 1.0
     float thrust_curve_expo = constrain_float(_thrust_curve_expo, -1.0f, 1.0f);
-    if (fabsf(thrust_curve_expo) < 0.001) {
+    if (is_zero(thrust_curve_expo)) {
         // zero expo means linear, avoid floating point exception for small values
-        return thrust;
+        return _lift_max * thrust;
```

Comment:
I wonder if we are missing a check that _batt_voltage_filt is positive..

Reply:
There is a check for this.... did you mean `_lift_max`?!

Reply:
I have confirmed that there is no way for this to be negative and that it is properly checked. The only thing I noticed was there were a number of places in the motor library where we use <= or < instead of is_negative or !is_positive.